server/peer.go

```diff
@@ -408,19 +408,25 @@ func (peer *Peer) useHistoryServer(tlsBaseConfig *tls.Config, historyServerName
     peer.alertsURI = alertsURI
     peer.historyURI = historyURI
    
-    tlsConfig := *tlsBaseConfig
-    tlsConfig.InsecureSkipVerify = noValidate
-    tlsConfig.ServerName = historyServerName
-    tlsConfig.RootCAs = nil // ensures that this uses the default root CAs
-    
-    peer.httpHistoryClient = &http.Client{ Transport: &http.Transport{ TLSClientConfig: &tlsConfig } }
+    if(tlsBaseConfig != nil) {
```

Comment:
You can get rid of these conditionals by adding one check at the top that sets `tlsBaseConfig` to an empty `tls.Config` object if it's nil. You only need one place in the code that actually constructs `httpAlertsClient` or `httpHistoryClient`

Reply:
Got it. Will do that.